pioneer/packages/app-tokenomics/src/lib/getValidatorData.ts

```diff
@@ -0,0 +1,51 @@
+import { ApiPromise } from '@polkadot/api';
+import { Vec } from '@polkadot/types';
+import { AccountId, IndividualExposure } from '@polkadot/types/interfaces';
+
+const getValidators = async (api: ApiPromise): Promise<any> => {
+  let [totalValidatorStake, slotStake, numberOfNominators, numberOfValidators] = [0, 0, 0, 0];
+  const validatorIds = await api.query.staking.currentElected() as Vec<AccountId>;
+  await Promise.all(validatorIds.map(async (id: AccountId, index: number) => {
+    const nominators = (await api.derive.staking.info(id)).stakers?.others as Vec<IndividualExposure>;
+    const totalStake = (await api.derive.staking.info(id)).stakers?.total.toNumber() as number;
```

Comment:
It seems that (accoring to TypeScript) `.stakers` can be `undefined` here, so perhaps a better way to deal with it would be:
```
    const nominators = (await api.derive.staking.info(id)).stakers?.others.toArray() || [];
    const totalStake = (await api.derive.staking.info(id)).stakers?.total.toNumber() || 0;
```